src/migrations/20231127000000-objectiveusers.js

```diff
@@ -0,0 +1,355 @@
+          null::JSONB "linkBack"
+        FROM "ZALObjectives" zg
+        LEFT JOIN "Users" u
+        ON zg.dml_as = u.id
+        JOIN "Objectives" g
+        ON zg.data_id = g.id
+        WHERE dml_as NOT IN (-1, 0) -- default and migration files
+        AND dml_type = '${dmlType}'
+        AND new_row_data -> 'title' IS NOT NULL
+        GROUP BY 1,2
+        ORDER BY 1,2
+        `,
+        typeName,
+      );
+
+      const collectObjectiveCollaboratorsViaActivityReport = () => collectObjectiveCollaborators(
+        /* sql */`
+        SELECT
+          g.id "objectiveId",
+          (ARRAY_AGG(ar."userId" ORDER BY ar.id ASC))[1] "userId",
+          MIN(ar."createdAt") "createdAt",
+          MIN(ar."createdAt") "updatedAt",
+          null::JSONB "linkBack"
+        FROM "Objectives" g
+        LEFT JOIN "ZALObjectives" zg
+        ON g.id = zg.data_id
+        LEFT JOIN "ActivityReportObjectives" arg
+        ON g.id = arg."objectiveId"
+        LEFT JOIN "ActivityReports" ar
+        ON arg."activityReportId" = ar.id
+        WHERE (zg.id IS NULL
+        OR (zg.dml_as IN (-1, 0)
+          AND zg.dml_type = 'INSERT'))
+        AND g."createdVia" IN ('activityReport')
+        GROUP BY 1
+        HAVING (ARRAY_AGG(ar."userId" ORDER BY ar.id ASC))[1] IS NOT NULL
+        AND MIN(ar."createdAt") IS NOT NULL
+        ORDER BY 1,2
+        `,
+        OBJECTIVE_COLLABORATORS.CREATOR,
+      );
+
+      const collectObjectiveCollaboratorsAsLinkers = () => collectObjectiveCollaborators(
+        /* sql */`
+       SELECT
+        "objectiveId",
+        "userId",
+        MIN("createdAt") "createdAt",
+        MAX("updatedAt") "updatedAt",
+        jsonb_build_object('activityReportIds', ARRAY_AGG(DISTINCT ar."activityReportId")) "linkBack"
+      FROM (
+        SELECT
+        arg."objectiveId" "objectiveId",
+        zarg.dml_as "userId",
+        MIN(arg."createdAt") "createdAt",
+        MAX(arg."createdAt") "updatedAt",
+          ARRAY_AGG(DISTINCT arg."activityReportId") "activityReportIds"
+        FROM "ZALActivityReportObjectives" zarg
+        LEFT JOIN "ZALActivityReportObjectives" zargd
+        ON zarg.data_id = zargd.data_id
+        AND zarg.id < zargd.id
+        AND zarg.dml_type = 'INSERT'
+        AND zargd.dml_type = 'DELETE'
+        JOIN "ActivityReportObjectives" arg
+        ON zarg.data_id = arg.id
+        WHERE zargd.id IS NULL
+        AND zarg.dml_type = 'INSERT'
+        AND zarg.dml_as NOT IN (-1, 0)
+        GROUP BY 1,2
+        UNION
+        SELECT
+        arg."objectiveId",
+        ar."userId",
+        MIN(arg."createdAt") "createdAt",
+        MAX(arg."createdAt") "updatedAt",
+          ARRAY_AGG(DISTINCT arg."activityReportId") "activityReportIds"
+        FROM "ActivityReportObjectives" arg
+        JOIN "ActivityReports" ar
+        ON arg."activityReportId" = ar.id
+        LEFT JOIN "ZALActivityReportObjectives" zarg
+        ON arg.id = zarg.data_id
+        AND zarg.dml_type = 'INSERT'
```

Comment:
Like the other comment, I think we want to filter out cases where the dml_as is in -1 and 0 so we don't miss cases where pre-audit log ARs get AROs via migration.